### PR TITLE
feat(v5): drop form row and form inline support

### PR DESCRIPTION
**Pull request by bpas247**
Drops support for the form-row and form-inline classes, and
removes any doc references associated with either of them.
https://v5.getbootstrap.com/docs/5.0/migration/#forms

### CODE REVIEW

**Comment by kyletsang** [comment]
I don't see the removal of the `inline` prop in Form.tsx

**Comment by bpas247** [comment]
> I don't see the removal of the `inline` prop in Form.tsx

yeah and I also noticed I didn't clean up all of the references of both features, so I ended up going through everything and cleaned up everything I found.

The new diffs were much larger so I ended up rebasing and splitting it across two different commits.

**Comment by jquense** [comment]
nice work folks, if it's not too much trouble for these PR's could we drop a reference link to the upstream changelog or src that inspired the change, should make it a bit easier for reviewers

File: src/Form.tsx [hunk 1]
```diff
@@ -91,7 +75,7 @@ const FormImpl: Form = (React.forwardRef(
         className={classNames(
           className,
           validated && 'was-validated',
-          inline && `${bsPrefix}-inline`,
+          bsPrefix,
```
Comment (kyletsang) [inline src/Form.tsx:78]: I think we can drop `bsPrefix` here?  I looked at the v5 docs and there is no `.form` class usage.

Comment (bpas247) [reply src/Form.tsx:78]: Hmm, why was the `form` class dropped? Also, if it was dropped, should we just drop our Form component entirely?

I have noticed that a lot of other component libraries (e.g. MUI and Chakrai-UI) don't have an overall Form component, so I'm curious if we should carry that trend over to our library as well.

Comment (kyletsang) [reply src/Form.tsx:78]: Took a closer look at bootstrap docs and it looks like there wasn't a `.form` class in 4.0 either.  I've seen usage of `.form-inline` and `.was-validated`.  

https://getbootstrap.com/docs/4.0/components/forms/

So in v5 with inline being dropped, this just leaves `.was-invalidated` to handle.  I guess it's nice to have Form around even if it's to handle 1 class... unsure :)

Comment (bpas247) [reply src/Form.tsx:78]: Either way the diffs for dropping support for the Form component would warrant it to be in a separate PR altogether, so we should open a new issue and discuss dropping support there.

In the meantime I'll drop `bsPrefix` and get this merged 👍

**Comment by jquense** [comment]
Given the heavier reliance on utility classes in v5 we should probably come up with an answer for #3029 and friends, since more and more of the "public" api of these components is shifting to these utils

**Pull request merged**